src/parse/parse_step.go

```diff
@@ -166,6 +170,40 @@ func activateTarget(tid int, state *core.BuildState, pkg *core.Package, label, d
 
 // parsePackage performs the initial parse of a package.
 func parsePackage(state *core.BuildState, label, dependent core.BuildLabel, subrepo *core.Subrepo) (*core.Package, error) {
+	packageName := label.PackageName
```

Comment:
are we duplicating all this code?

Reply:
Yeah, I think this makes the v16 PR very straight forward. Do you not like this approach?

Reply:
Just a bit worried about how long we're going to keep both side by side for. I guess we can remember to update both if we need to change one...

Reply:
Definitely a concern. Ideally we have tests that will make this obvious. Looking at the git history, this file is updated once every two or so months. That seems like it should be okay.